### PR TITLE
CASMINST-4471 fix nexus init/setup to work with newer podman version

### PR DESCRIPTION
**Pull request by jacobsalmela**
## Summary and Scope

Fixes the nexus init script.  `nexus` is now `nexus3`.  This also removed the `-d` flag in the `podman` command since it is no longer valid syntax.

## Issues and Related PRs

* Resolves CASMINST-4471
* Merge with https://github.com/Cray-HPE/metal-basecamp/pull/12

## Testing

Manually tested these changes and I was able to start nexus.

### Tested on:

  * `#redbull`

## Risks and Mitigations

Low, it's broken now.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: Jenkinsfile.github [hunk 1]
```diff
@@ -13,37 +39,48 @@ pipeline {
...
             agent {
                 docker {
-                    image 'artifactory.algol60.net/csm-docker/stable/sle15sp3_build_environment'
+                    label 'docker'
+                    image "artifactory.algol60.net/csm-docker/stable/sle15sp3_build_environment:latest"
                     reuseNode true
                     registryUrl 'https://artifactory.algol60.net/'
                     registryCredentialsId 'artifactory-algol60'
                     // Run as root
-                    args "-u root -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/usr/bin/docker --group-add 999"
```
Comment (manderson-hpe) [inline Jenkinsfile.github]: I'm curious why this is not needed anymore.

**Comment by manderson-hpe** [comment]
Reviewed it. Seems like a fine update. I wasn't able to find anything that stood out as incorrect.

File: Jenkinsfile.github (hunk 1 above)
Comment (rustydb) [reply Jenkinsfile.github]: Does it have to do with these two lines?
```
                    label 'docker'
                    args "-u root"
```

File: systemd/nexus-init.sh [hunk 2]
```diff
@@ -58,10 +78,9 @@ if ! podman inspect "$NEXUS_CONTAINER_NAME" ; then
         --conmon-pidfile "$NEXUS_PIDFILE" \
         --cidfile "$NEXUS_CIDFILE" \
         --cgroups=no-conmon \
-        -d \
```
Comment (rustydb) [inline systemd/nexus-init.sh:61]: was this argument deprecated, or do we no longer run this as detached?

**Review by rustydb — APPROVED** [review]

File: systemd/nexus-init.sh (hunk 2 above)
Comment (jacobsalmela) [reply systemd/nexus-init.sh:61]: Yes, deprecated.  The same change is in the basecamp init file

File: systemd/nexus-init.sh [hunk 3]
```diff
@@ -58,10 +78,9 @@ if ! podman inspect "$NEXUS_CONTAINER_NAME" ; then
...
         --network host \
         --volume "${NEXUS_VOLUME_NAME}:${NEXUS_VOLUME_MOUNT}" \
         --name "$NEXUS_CONTAINER_NAME" \
         "$NEXUS_IMAGE" || exit
     podman inspect "$NEXUS_CONTAINER_NAME" || exit
-fi
+fi
```
Comment (dborman-hpe) [inline systemd/nexus-init.sh]: Put back the final newline!

File: systemd/nexus-setup.sh [hunk 4]
```diff
@@ -85,4 +105,4 @@ EOF
 
 nexus-repositories-create /tmp/nexus-repositories.yaml >&2
 nexus-enable-docker-realm >&2
-" || exit
+" || exit
```
Comment (dborman-hpe) [inline systemd/nexus-setup.sh]: Put back the final newline!

File: Jenkinsfile.github (hunk 1 above)
Comment (jacobsalmela) [reply Jenkinsfile.github]: Fixed

**Review by heemstra — APPROVED** [review]

**Pull request merged**